e2e/accessibility.ts

```diff
@@ -45,69 +35,106 @@ const main = async (): Promise<void> => {
     urls.add(url);
   }
 
+  await browser.close();
+
+  return urls;
+};
+
+const main = async (): Promise<void> => {
+  assert(
+    fs.existsSync(DIST_PATH) && fs.existsSync(INDEX_HTML),
+    'Missing dist/'
+  );
+
+  const app = express();
+  app.use(express.static(DIST_PATH));
+  app.use((req, res) => res.sendFile(INDEX_HTML));
+  const server = app.listen(0);
+  const { port } = server.address() as AddressInfo;
+  const urls = await getComponentUrls(port);
+
+  // Allow multiple instances of puppeteer to run to speed up testing
+  const queue = new PQueue({
+    concurrency: CONCURRENCY
+  });
+
   // Analyze each URL.
-  for (const url of urls) {
-    const component = (url.split('/').pop() as string) || 'Index';
-
-    await page.goto(url);
-
-    for (const theme of THEMES) {
-      await page.evaluate(
-        theme => {
-          document.body.className = '';
-          document.body.classList.add(`cauldron--theme-${theme}`);
-        },
-        [theme]
-      );
-
-      // Try to wait until there is an idle period up to a max of 5s
-      await Promise.race([
-        await page.evaluate(
-          () =>
-            new Promise(resolve =>
-              // Typescript does not implement experimental apis but this should exist within puppeteer
-              // see: https://github.com/microsoft/TypeScript/issues/21309
-              (window as any).requestIdleCallback(resolve)
-            )
-        ),
-        await page.waitForTimeout(5000)
-      ]);
-
-      const axe = new AxePuppeteer(page, AXE_SOURCE).withTags([
-        'wcag2a',
-        'wcag2aa',
-        'wcag21a',
-        'wcag21aa'
-      ]);
-      const { violations } = (await axe.analyze()) as AxeResults;
-
-      let symbol = logSymbols.success;
-      if (violations.length) {
-        symbol = logSymbols.warning;
-        foundViolations = true;
-      }
-
-      const title =
-        chalk.cyan(chalk.bold(component)) + ' (' + chalk.italic(theme) + ')';
-      const dots = '.'.repeat(MAX_WIDTH - component.length - symbol.length);
-      console.log(title, dots, symbol);
-
-      for (const { id, help, nodes } of violations) {
-        console.log('↳', chalk.underline(chalk.magenta(id)), '━', help);
-        nodes.map(node => {
-          console.log(
-            '  ↳',
-            chalk.bgGreen(chalk.black(`${node.target}`)),
-            chalk.yellowBright(node.html)
+  await Promise.all(
```

Comment:
Changes here were primarily wrapping the puppeteer browse spin-up into a queue to run in paralle.